packages/amplify-provider-awscloudformation/src/utils/remove-dependent-function.ts

```diff
@@ -0,0 +1,68 @@
+import { $TSAny, $TSContext, $TSObject, pathManager } from 'amplify-cli-core';
+import path from 'path';
+import * as TransformPackage from 'graphql-transformer-core';
+
+export async function removeDependencyOnFunctions(
```

Comment:
I think a better name for this function is `ensureValidFunctionModelDependencies` because this function may not remove anything (if no models have been removed or if the customer selects no) and it's not the models that depends on the functions, it's the functions that depend on the models